### PR TITLE
Sergei Palmer Airport Challenge Updated

### PR DESCRIPTION
**Pull request by s-palmer**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require_relative 'plane'
+require_relative 'weather'
+
+class Airport
+  DEFAULT_CAPACITY = 5
+  attr_accessor :capacity
```
Comment (Kntanos) [inline lib/airport.rb:6]: Is an attr-accessor required or attr_reader would be enough?

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def weather_station
+    # @weather = Weather.new
+    @weather.clear?
+  end
+
+  def takeoff(plane)
+    check_hangers
+    if weather_station
```
Comment (Kntanos) [inline lib/airport.rb:31]: Can be refactored with a ternary expression

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      puts "Skies are clear. Plane taking off."
+      airport.delete(plane)
+    else
+      puts "It's too stormy to fly"
+    end
+  end
+
+  def land(plane)
```
Comment (Kntanos) [inline lib/airport.rb:39]: Is not checking if the weather is not stormy.

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    fail "No space" if full?
+    @airport << plane
+    puts "Plane has landed"
```
Comment (Kntanos) [inline lib/airport.rb:42]: I like the return comments confirming the action.

File: lib/weather.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def clear?
+    case rand(1..4)
+    when 1..3
+      @weather = true
+    when 4
+      @weather = false
+    end
+    # 1..3 = good weather
```
Comment (Kntanos) [inline lib/weather.rb:15]: Good comment here.

File: lib/weather.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  attr_reader :weather
+
+  def initialize
+    @weather = weather
+  end
+
+  def clear?
+    case rand(1..4)
+    when 1..3
+      @weather = true
```
Comment (Kntanos) [inline lib/weather.rb:11]: Since you have an attr_reader for @weather, you don't need to keep using instance variables

File: lib/weather.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    when 4
+      @weather = false
```
Comment (Kntanos) [inline lib/weather.rb:13]: Since you have an attr_reader for @weather, you don't need to keep using instance variables

File: lib/airport.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
+require_relative 'plane'
+require_relative 'weather'
+
+class Airport
+  DEFAULT_CAPACITY = 5
+  attr_accessor :capacity
+  attr_reader :airport
```
Comment (Kntanos) [inline lib/airport.rb:7]: Add :weather to attr_reader, so you don't need to use instance variables in the rest of the code

File: lib/airport.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    @airport.size >= capacity
+  end
+
+  def check_hangers 
+    fail "No Planes." if @airport.empty?
+    puts "Plane ready for take-off"
+  end
+
+  def weather_station
+    # @weather = Weather.new
```
Comment (Kntanos) [inline lib/airport.rb:25]: Good one explaining what this is doing

File: lib/airport.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+require_relative 'weather'
+
+class Airport
+  DEFAULT_CAPACITY = 5
+  attr_accessor :capacity
+  attr_reader :airport
+
+  def initialize(capacity = DEFAULT_CAPACITY)
+    @capacity = capacity
+    @airport = []
```
Comment (Kntanos) [inline lib/airport.rb:11]: Why not call this variable hangers rather than airport? (I thing it's hangar)

File: lib/weather.rb (hunk 7 above)
Comment (s-palmer) [reply lib/weather.rb:13]: Done thanks  :)

File: lib/airport.rb (hunk 8 above)
Comment (s-palmer) [reply lib/airport.rb:7]: Done thanks!

File: lib/airport.rb (hunk 3 above)
Comment (s-palmer) [reply lib/airport.rb:39]: Not yet!

File: README.md [hunk 11]
```diff
@@ -60,30 +47,49 @@ To ensure safety
...
 
-Your task is to test drive the creation of a set of classes/modules to satisfy all the above user stories. You will need to use a random number generator to set the weather (it is normally sunny but on rare occasions it may be stormy). In your tests, you'll need to use a stub to override random weather to ensure consistent test behaviour.
-
-Your code should defend against [edge cases](http://programmers.stackexchange.com/questions/125587/what-are-the-difference-between-an-edge-case-a-corner-case-a-base-case-and-a-b) such as inconsistent states of the system ensuring that planes can only take off from airports they are in; planes that are already flying cannot take off and/or be in an airport; planes that are landed cannot land again and must be in an airport, etc.
-
-For overriding random weather behaviour, please read the documentation to learn how to use test doubles: https://www.relishapp.com/rspec/rspec-mocks/docs . There’s an example of using a test double to test a die that’s relevant to testing random weather in the test.
+## How to use
 
-Please create separate files for every class, module and test suite.
+```
```
Comment (leoht) [inline README.md:52]: Nice detailed README with the instructions!

**Review by leoht — COMMENTED** [review]
Nice work! The code is quite readable and clean, I mainly commented on things like:
 * Instance variables and their use or purpose — they're sometimes not needed when you don't need to store a state on the object
 * Plain strings in the code printed by `puts` — this is OK to try out things in the terminal, but be mindful of having them with your "main logic" code, since printing out things deals with presenting or displaying info to the user, so it probably deserves to be in its own class or component, to respect the SRP
 * Some minor things about refactoring, renaming, simplifying, etc :)

Let me know if some of the feedback is unclear and we can chat!
